### ISSUE TITLE
`min_type_alias_impl_trait` doesn't report an error when impl Trait is used in a struct field

### ISSUE THREAD

**Issue by nikomatsakis** [rust-lang/rust#86732]
This example ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=25016f6db9b1562946cf6b5a968a5bd6)):

```rust
#![feature(min_type_alias_impl_trait)]
type X = impl Iterator<Item = u64> + Unpin;

struct Foo(X);
impl Foo {
    fn new(z: Vec<u64>) -> Self {
        Foo(z.into_iter())
    }
}
```

should report an error because the use of `X` in that position is not permitted with `min_type_alias_impl_trait`. But it does not.

(Spun out from https://github.com/rust-lang/rust/issues/64445)

**Comment by nikomatsakis** [comment]
Also we should test in enum fields

```rust
#![feature(type_alias_impl_trait)]

type Foo = impl Copy;

enum Wrapper<T> {
    First(T),
    Second
}

fn produce() -> Wrapper<Foo> {
    Wrapper::Second
}
```

**Comment by oli-obk** [comment]
I think we should just keep allowing this.
